graylog2-server/src/main/java/org/graylog2/web/resources/WebInterfaceAssetsResource.java

```diff
@@ -0,0 +1,125 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog2.web.resources;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.graylog2.plugin.Plugin;
+import org.graylog2.web.IndexHtmlGenerator;
+import org.graylog2.web.ModuleManifest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+
+@Path("{filename: .*}")
+public class WebInterfaceAssetsResource {
+    private static final Logger LOG = LoggerFactory.getLogger(WebInterfaceAssetsResource.class);
+
+    private static String pathPrefix = "web-interface/assets";
+    private static String pluginPathPrefix = "/plugin/";
+    private static String manifestFilename = "module.json";
+    private final IndexHtmlGenerator indexHtmlGenerator;
+    private final ObjectMapper objectMapper;
+
+    @Inject
+    public WebInterfaceAssetsResource(ObjectMapper objectMapper,
+                                      Set<Plugin> plugins) throws IOException {
+        this.objectMapper = objectMapper;
+        final List<String> jsFiles = new ArrayList<>();
+        final List<String> cssFiles = new ArrayList<>();
+
+        plugins.stream().forEach(plugin -> {
```

Comment:
This is being done on every request because the jersey resource should be constructed on every request. Can we move loading files from plugin's resources into a common plugin class somehow?